setup.cfg

```diff
@@ -1,5 +1,5 @@
 [tool:pytest]
-addopts = -vs --reuse-db --showlocals --tb=short
+addopts = -vs --nomigrations --reuse-db --showlocals --tb=short
```

Comment:
Everything fails.  I tried playing around with combinations of `--reuse-db`, `--create-db `, `--nomigrations`, and `--migrations` and I think it's that migrations can't run without the database being created first.

`--migrations` when the database exists (e.g. locally with `--reuse-db`  after the first attempt that creates the database) works, though it's quite slow to start up .  Because travis is always running without an existing database it always fails.  Potentially we could have travis run with `--nomigrations` but local `pytest src/...` would still run migrations.

Reply:
>  Because travis is always running without an existing database it always fails.

How do you mean, travis fails when you're using the `--migrations`, even though pytest-django should create a database long before that? Or am I misunderstanding?

Reply:
`pytest --migrations` fails when there isn't a database already, correct.  (`--migrations` is the the default, so if you don't have `--nomigrations` in setup.cfg then you get that by default).

Locally that would be the equivalent of `pytest --create-db --migrations` or just deleting the database manually and running `pytest --reuse-db --migrations`. 

tbh I've not debugged the pytest|django|pytest-django source code to find out where/why it's happening.  I'm only guessing its caused by the absence of a database (as that's the difference between `--reuse-db` and not)  https://travis-ci.org/eviljeff/olympia/builds/584205012 is before I added the cfg change.

Reply:
Can't we keep the migrations but manually create the database in travis before running the tests ? It  should be almost instantaneous to run a `mysql -e 'CREATE DATABASE olympia'` (or whatever the database name is) so we could just put that in `before_script` in `.travis.yml`

Reply:
We are already doing that actually, https://github.com/mozilla/addons-server/blob/a689d07f31e8a83e9e5b1666a94e1572cf6f1949/.travis.yml#L53

Reply:
Eh yeah I'm blind. So the database should exist already, so I don't see why migrations fail...

Reply:
> Also note that the test database would be test_olympia_gw1 and test_olympia_gw0, not sure if that affects things.

Indeed, so the part I linked is actually quite useless. Maybe we should remove it then

Reply:
Looked more into this.  It's not (as I naively assumed) the absence or presence of the database - it's the translations testapp models.  They contain a foreign key to the Translation model, and foreign keys from models that _don't_ have migrations (e.g. testapp.TranslatedModel) to models that _do_ have migrations (e.g. Translation) are unsupported.

If I rename models.py to something else then the tests in general pass - though most of the translations tests don't because they reference the testapp models which now didn't have their database tables created.

In theory we could generate migrations for the testapp models too.  But I've not worked out how to reference them in `manage.py makemigrations xxx` (i.e. what is `xxx` for the testapp app)

Reply:
https://github.com/mozilla/addons-server/pull/12349/commits/35b1260c4f3df07a39b9743d043ff5567473cacf has fixed it.  I added the testapp to `INSTALLED_APPS` in settings_base.py to allow `makemigrations testapp` to work.